.gitlab/docker/Dockerfile.spack_compiler

```diff
@@ -0,0 +1,25 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+SHELL ["/bin/bash", "-c"]
+
+# Disable host compatibility to be able to compile for broadwell on zen2
```

Comment:
I should perhaps been more specific about this comment: this will be compiling for other architectures in the future. Instead of having a comment about which architectures we're targeting, the comment should explain _why_ the architecture will/may be different (builds are done on a different system from where tests are run).